scrapy_webarchive/downloadermiddlewares.py

```diff
@@ -57,8 +57,10 @@ def process_request(self, request: Request, spider: Spider):
             self.stats.inc_value("webarchive/response_not_found", spider=spider)
             return Response(url=request.url, status=404)
         
+        request.meta["warc_record_response"] = warc_record
+        
```

Comment:
I think it's a little strange to store the _response_ info the the _request_ meta. Why not the response meta?

Reply:
I agree that it would look more logical to apply this to the `response.meta`, however under the hood the are exactly the same thing. Look at the property `meta` of the `scrapy.http.response.Response` class:

```python
@property
def meta(self):
    try:
        return self.request.meta
    except AttributeError:
        raise AttributeError(
            "Response.meta not available, this response "
            "is not tied to any request"
        )
```

Applying to either one will give the same result. But perhaps if it might ever change it might be more safe to apply this to the `Response` object not the `Request` object, so I will change this.

Reply:
That's btw also why it is named `warc_record_response`. It will also be available on the request and therefore more be confusing to call it `warc_record`.

Reply:
Hah, thanks for explaining! :+1: